### PR TITLE
Create listing details page

### PR DESCRIPTION
**Pull request by henryhu1**
- New page for listing details
  - Added nav path
- Use listing details GET endpoint
- Use listing image GET endpoint
- Populate page with data
  - Image carousel

### CODE REVIEW

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageUiState.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageUiState.kt]: Why is this nullable?
Is there any case where `ListingDetails` will be null when the page is `Loaded`?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt]: I think you can just remove this parameter?
You're using `SavedStateHandle`, right?

File: android/app/src/main/java/org/uwaterloo/subletr/theme/SubletrTypography.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/theme/SubletrTypography.kt]: Are you sure this isn't used anywhere else?
If it is used somewhere else, it might mess up another screen.

If you need a new text style, just make a new variable instead.

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageUiState.kt
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageUiState.kt]: honestly I was thinking this had to match with stream in the viewmodel having `Optional`
but other than that no reason for why it's nullable, I'll fix it

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt]: yeah just found out it _can_ be removed

File: android/app/src/main/java/org/uwaterloo/subletr/theme/SubletrTypography.kt
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/theme/SubletrTypography.kt]: I'm sure this wasn't used anywhere else BUT I agree another typography style is better

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageUiState.kt
Comment (leeminkyu16) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageUiState.kt]: Why does `listingDetailsStream` need to be `Optional`?
The default value that you pass in when you call `subscribeAsState` should be sufficient, no? (especially since the values don't change).

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt:57]: I'm pretty sure that this line basically never runs

Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt]: I don't think this ever runs (unless I'm actually losing my mind)?
There is no case where listingDetailsStream ever has the value of `Optional.empty()`

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt:46]: Maybe `subscribeOn(Schedulers.computation())`?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt]: Also, maybe we should make a global function for this, like `fun String.base64ToBitmap(): Bitmap {...}`

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt:398]: Technically bad practice but we can keep it for now since we have a demo coming up.
Can you just add a TODO to make this localized in the future? (Not all languages use "MM/dd/yyyy")?

For future reference, it seems like changing the format thing with `DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG)` might work?

File: android/app/src/main/java/org/uwaterloo/subletr/navigation/NavigationDestination.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/navigation/NavigationDestination.kt]: If you want the "home" icon to be highlighted on the bottom bar, I think you need this to be "home/listingdetails"? 

I changed the logic to check the substring before the first "/" when looking to see what to highlight in the bottombar when I "fixed" navigation for the third time or whatever.

File: android/app/src/main/java/org/uwaterloo/subletr/navigation/MainNavigation.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/navigation/MainNavigation.kt]: Wait, sorry, isn't this an int?

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt]: indeed you're not losing your mind, removed this

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt]: yeah made a utils package instead with `ImageUtils`

File: android/app/src/main/java/org/uwaterloo/subletr/navigation/MainNavigation.kt
Comment (henryhu1) [reply android/app/src/main/java/org/uwaterloo/subletr/navigation/MainNavigation.kt]: yeah I changed it
the id being `-1908663237` made me think it had to be a string

File: android/app/src/main/java/org/uwaterloo/subletr/theme/SubletrTypography.kt
Comment (leeminkyu16) [inline android/app/src/main/java/org/uwaterloo/subletr/theme/SubletrTypography.kt:132]: I think we do need to change all of these to `secondaryTextColor` at some point but we can do that later ^-^

**Review by leeminkyu16 — APPROVED** [review]
Let's Gamble, Try Merging

File: android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt
Comment (andr-dev) [reply android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt:398]: _localization_

**Review by leeminkyu16 — APPROVED** [review]

**Review by andr-dev — CHANGES_REQUESTED** [review]
Hey, just putting a block on this until we get rid of the merge commit.

**Review by andr-dev — APPROVED** [review]
Nice work!

**Pull request merged**
